src/blocks/checkout-button/view.php

```diff
@@ -112,6 +113,12 @@ function render_callback( $attributes ) {
 			$frequency = \WC_Subscriptions_Product::get_period( $product );
 		}
 
+		$product_type = 'product';
+		// Update the product type.
+		if ( method_exists( 'Newspack_Blocks\Tracking\Data_Events', 'get_product_type' ) ) {
```

Comment:
Unlike the defensive checks for `\Newspack\` methods or global functions defined by WooCommerce, there's no real reason we need to check for the existence of these methods since they're defined by this plugin (so if this code is running, we can be certain that the other code in the plugin exists too).

```suggestion
		$product_type = \Newspack_Blocks\Tracking\Data_Events::get_product_type( $product_id );
```

Reply:
Makes sense! This has been removed, plus any others for methods coming from this plugin -- thanks!